### PR TITLE
SecurityPkg/Tcg/Opal/OpalPassword: check for both BLOCK_IO and BLOCK_IO2

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c [hunk 1]
```diff
@@ -2793,43 +2802,70 @@ OpalEfiDriverBindingStart(
...
+    BlkIo = NULL;
+
   if (EFI_ERROR(Status)) {
-    //
-    // Block_IO not supported on handle
-    //
-    if(Status == EFI_UNSUPPORTED) {
-      BlkIo = NULL;
-    } else {
+    Status2 = gBS->OpenProtocol(
```
Comment (mkopec) [inline SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c:2810]: Should EFI_BLOCK_IO2_PROTOCOL not take priority over EFI_BLOCK_IO_PROTOCOL?

**Review by mkopec — COMMENTED** [review]
One question from me

File: SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c (hunk 1 above)
Comment (miczyg1) [reply SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c:2810]: UEFI spec does not specify which protocol has priority. I have checked all drivers that produce BLOCK_IO or BLOCK_IO2 and it looks like all drivers produce both, so it shouldn't matter. All we need the BLOCK_IO for is the MediaId, which shouldn't be different depending on BLOCK_IO or BLOCK_IO2 @mkopec

Comment (mkopec) [reply SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c:2810]: Ok, thanks.

**Review by mkopec — APPROVED** [review]

**Pull request merged**